aten/src/ATen/native/cpu/UpSampleKernelAVXAntialias.h

```diff
@@ -331,62 +355,75 @@ void upsample_avx_bilinear_uint8(
   // tensors and just copy part of them (line by line).
   for (const auto i : c10::irange(batch_size)) {
 
-    at::Tensor unpacked_input = (is_rgba) ? input[i] : unpack_rgb(input[i]);
+    at::Tensor unpacked_input = (is_rgb_or_rgba) ? input[i] : unpack_rgb(input[i]);
     at::Tensor unpacked_output;
 
     if (need_horizontal) {
+      at::Tensor unpacked_output_temp = (is_rgb_or_rgba && !need_vertical) ? output[i] : buffer_horiz;
 
-      at::Tensor unpacked_output_temp = (is_rgba && !need_vertical) ? output[i] : buffer_horiz;
-
-      ImagingResampleHorizontal(
+      if (is_rgb_or_rgba && num_channels == 3) {
+        ImagingResampleHorizontal<3>(
           unpacked_output_temp,
           unpacked_input,
           ksize_horiz,
           horiz_indices_weights,
           horiz_weights_precision);
+      } else {
+        ImagingResampleHorizontal<4>(
```

Comment:
Shouldn't we also check for `is_rgb_or_rgba` here, like we did above? I.e. what if `is_rgb_or_rgba` is false here: is this branch still doing the right thing?